pallets/nfts/src/tests.rs

```diff
@@ -1942,53 +2330,307 @@ fn approving_multiple_accounts_works() {
 			account(5),
 			Some(2)
 		));
-		assert_eq!(
-			approvals(0, 42),
-			vec![(account(3), None), (account(4), None), (account(5), Some(current_block + 2))]
+		assert_eq!(
+			approvals(0, 42),
+			vec![(account(3), None), (account(4), None), (account(5), Some(current_block + 2))]
+		);
+
+		assert_ok!(Nfts::transfer(RuntimeOrigin::signed(account(4)), 0, 42, account(6)));
+		assert_noop!(
+			Nfts::transfer(RuntimeOrigin::signed(account(3)), 0, 42, account(7)),
+			Error::<Test>::NoPermission
+		);
+		assert_noop!(
+			Nfts::transfer(RuntimeOrigin::signed(account(5)), 0, 42, account(8)),
+			Error::<Test>::NoPermission
+		);
+	});
+}
+
+#[test]
+fn approvals_limit_works() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			account(1),
+			default_collection_config()
+		));
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(account(1)),
+			0,
+			42,
+			account(2),
+			default_item_config()
+		));
+
+		for i in 3..13 {
+			assert_ok!(Nfts::approve_transfer(
+				RuntimeOrigin::signed(account(2)),
+				0,
+				42,
+				account(i),
+				None
+			));
+		}
+		// the limit is 10
+		assert_noop!(
+			Nfts::approve_transfer(RuntimeOrigin::signed(account(2)), 0, 42, account(14), None),
+			Error::<Test>::ReachedApprovalLimit
+		);
+	});
+}
+
+#[test]
+fn approve_collection_transfer_works() {
```

Comment:
Please add the origin checks here too.

Reference: https://github.com/r0gue-io/pop-node/blob/3476994d74bb76ac2ed58e091133a89599c18d28/pallets/api/src/fungibles/tests.rs#L85-L87

Reply:
Resolved by adding the origin checks for `approve_collection_transfer_works`, `cancel_collection_approval_works` and `clear_collection_approvals_works`